lib/tokens.ts

```diff
@@ -0,0 +1,178 @@
+import { PublicKey } from "@solana/web3.js";
+import { TokenProps } from "./types";
+import {
+  JsonMetadata,
+  MPL_TOKEN_METADATA_PROGRAM_ID,
+  deserializeMetadata,
+} from "@metaplex-foundation/mpl-token-metadata";
+import { RpcAccount } from "@metaplex-foundation/umi";
+import { Provider } from "@coral-xyz/anchor";
+import { USDCAddress, USDCMetadata } from "./constants";
+import { Mint, TOKEN_2022_PROGRAM_ID, getMint } from "@solana/spl-token";
```

Comment:
We can also use get getTokenMetadata() from this import and try that with the TOKEN_PROGRAM_ID this before we just end up truncating.